### PR TITLE
Drop protocol support

### PR DESCRIPTION
**Pull request by albertandrejev**
This PR adds support of the Drop protocol swap https://github.com/hadronlabs-org/drop-contracts

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ dexter-stable-pool   = "1.1.1"
 dexter-weighted-pool = "1.1.1"
 dexter-lp-token  = "1.0.0"
 dexter-router    = "1.1.0"
-cosmwasm-schema  = "1.5.0"
-cosmwasm-std     = { version = "1.5", features = ["stargate"] }
+cosmwasm-schema  = "1.5.4"
+cosmwasm-std     = { version = "1.5.4", features = ["stargate", "cosmwasm_1_2"] }
```
Comment (NotJeremyLiu) [inline Cargo.toml:31]: @albertandrejev Does this break our contracts being able to be deployed on chains without cosmwasm 1.2?

Comment (NotJeremyLiu) [reply Cargo.toml:31]: @albertandrejev why do we need it, can we remove?

Comment (albertandrejev) [reply Cargo.toml:31]: let me check it

Comment (albertandrejev) [reply Cargo.toml]: removed

File: contracts/adapters/swap/drop/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        _ => unimplemented!(),
+    }
+}
+
+fn get_exchange_rate(deps: Deps) -> ContractResult<Decimal> {
+    let drop_core_contract_address = DROP_CORE_CONTRACT_ADDRESS.load(deps.storage)?;
+
+    let exchange_rate: Decimal = deps.querier.query_wasm_smart(
+        drop_core_contract_address,
+        &drop_staking_base::msg::core::QueryMsg::ExchangeRate {},
```
Comment (NotJeremyLiu) [inline contracts/adapters/swap/drop/src/contract.rs:316]: @albertandrejev How accurate is this for arbitrary bond sizes? Let's say exchange rate returns 1:1, will the realized exchanged amount if put in 1 unit of X vs 1_000_000_000 units of X be the same? If not, is there a way to simulate the actual outcome which takes into account bond size?

File: Cargo.toml (hunk 1 above)
Comment (NotJeremyLiu) [reply Cargo.toml]: Awesome, thanks

File: contracts/adapters/swap/drop/src/contract.rs (hunk 2 above)
Comment (albertandrejev) [reply contracts/adapters/swap/drop/src/contract.rs:316]: Yes, exchange rate is linear in frames of one transaction, exchanged amount will be with same exchange rate inside  one message.

Comment (albertandrejev) [reply contracts/adapters/swap/drop/src/contract.rs:316]: No need to simulate

Comment (NotJeremyLiu) [reply contracts/adapters/swap/drop/src/contract.rs:316]: Awesome, thank you!

File: contracts/adapters/swap/drop/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    // Enforce the caller is the entry point contract
+    if info.sender != entry_point_contract_address {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Get coin in from the message info, error if there is not exactly one coin sent
+    let coin_in = one_coin(&info)?;
+
+    let bridged_denom = BRIDGED_DENOM.load(deps.storage)?;
+    let canonical_denom = CANONICAL_DENOM.load(deps.storage)?;
```
Comment (wesl-ee) [inline contracts/adapters/swap/drop/src/contract.rs:148]: This is vocabulary that's re-used from the lido satellite contract, is this name still accurate here or should it be `REMOTE_DENOM` and `BONDED_DENOM` or similar?

Comment (albertandrejev) [reply contracts/adapters/swap/drop/src/contract.rs:148]: Yes, REMOTE_DENOM and BONDED_DENOM are better, I will fix

**Review by wesl-ee — APPROVED** [review]

**Pull request merged**